nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-api/src/main/java/org/nuxeo/ecm/directory/BaseSession.java

```diff
@@ -443,6 +443,20 @@ public void deleteEntry(String id) throws DirectoryException {
         getDirectory().invalidateCaches();
     }
 
+    /**
+     * @param results the query results without limit and offet
+     * @param limit maximum number of results ignored if less than 1
+     * @param offset number of rows skipped before starting, will be 0 if less than 0.
+     * @return the result with applied liimit and offset
+     * @since 10.1
+     * @see {@link Session#query(Map, Set, Map, boolean, int, int)}
+     */
+    protected DocumentModelList applyQueryLimits(DocumentModelList results, int limit, int offset) {
+        int safeOffset = Math.max(0, offset);
```

Comment:
You can reassign to `offset`, we don't forbid assignment to method parameters and in this case it's more readable.

Reply:
> believe no one should.

Why? It's a feature of Java that it's allowed, and it makes for shorter code and avoids having a new local variable which the reader has to wrap its mind around.